From: Mike Snitzer <snitzer@redhat.com>
Date: Mon, 11 Jan 2010 21:11:35 -0500
Subject: [block] Honor the gfp_mask for alloc_page() in blkdev_issue_discard()
Message-id: <1263244295-5871-1-git-send-email-snitzer@redhat.com>
Patchwork-id: 22408
O-Subject: [RHEL6 PATCH] block: Honor the gfp_mask for alloc_page() in
	blkdev_issue_discard()
Bugzilla: 554719
RH-Acked-by: Pete Zaitcev <zaitcev@redhat.com>
RH-Acked-by: Vivek Goyal <vgoyal@redhat.com>
RH-Acked-by: Jeff Moyer <jmoyer@redhat.com>

BZ 548796

backport from linux-2.6:

commit e79e95db5cffb2e01170d510686489c40937faa1
Author: OGAWA Hirofumi <hirofumi@mail.parknet.co.jp>
Date:   Tue Dec 29 08:53:54 2009 +0100

    block: Honor the gfp_mask for alloc_page() in blkdev_issue_discard()

Signed-off-by: Mike Snitzer <snitzer@redhat.com>

diff --git a/block/blk-barrier.c b/block/blk-barrier.c
index 8873b9b..8618d89 100644
--- a/block/blk-barrier.c
+++ b/block/blk-barrier.c
@@ -402,7 +402,7 @@ int blkdev_issue_discard(struct block_device *bdev, sector_t sector,
 		 * our current implementations need.  If we'll ever need
 		 * more the interface will need revisiting.
 		 */
-		page = alloc_page(GFP_KERNEL | __GFP_ZERO);
+		page = alloc_page(gfp_mask | __GFP_ZERO);
 		if (!page)
 			goto out_free_bio;
 		if (bio_add_pc_page(q, bio, page, sector_size, 0) < sector_size)
